MCU_Aurix/bootloader/src/memory.c

```diff
@@ -38,15 +110,64 @@ uint8_t did_bl_write_end_add_core2[FBL_DID_BL_WRITE_END_ADD_CORE2_BYTES_SIZE]=
  * This function reads all the data from the memory and fills the variables (TBD)
  */
 void init_memory(void){
-    // TODO: Read from real memory and fill the variables
+
+    // Reading from memory
+    uint32_t len = sizeof(memData);
+    uint8_t *dataRead = flashRead((uint32)DID_DATA_FLASH_ADDR, len);
+
+    // Copy content to Variables
+    write_to_variable(len, dataRead, (uint8_t*)(&memData));
+    free(dataRead);
+
+    // Validate the Memory Based on Programming Date and System Name
+    init = validateMemory();
+
+    // Check Init
+    if(!init){
+        uint8_t did_system_name[] = FBL_DID_SYSTEM_NAME_DEFAULT;
+        write_to_variable(sizeof(did_system_name), did_system_name, memData.did_system_name);
+
+        uint8_t did_programming_date[] = FBL_DID_PROGRAMMING_DATE_DEFAULT;
+        write_to_variable(FBL_DID_PROGRAMMING_DATE_BYTES_SIZE, did_programming_date, memData.did_programming_date);
```

Comment:
warum hier sizeof(did_system_name) und sonst die makros?

Reply:
Der Name hat maximal 32 Byte Zeichen, welches das Makro zurückliefert. Der Default System Name ist aber kürzer, welcher durch sizeof bei einem char array korrekt ausgelesen wird. D.h. es werden nur ein Anteil der 32 Byte für den Namen verwendet bzw. in die Variable geschrieben.

Reply:
okay macht sinn